xc7/utils/prjxray_assign_tile_pin_direction.py

```diff
@@ -259,7 +261,8 @@ def main():
                 assert key not in edge_assignments, key
                 edge_assignments[key] = []
 
-    conn = sqlite3.connect(args.connection_database)
+    db_cache = DatabaseCache(args.connection_database)
```

Comment:
assign_tile_pin_direction is read-only, the cache should have no effect.

Reply:
You can also enforce read-only with:
https://www.sqlite.org/pragma.html#pragma_query_only

Reply:
Implemented the read-only option.